### PR TITLE
executors: Improve release process for binary

### PR DESCRIPTION
**Pull request by eseliger**
This does a few changes:
- Build binary and VM separately, it was a dirty hack that they ran in the same BK job
- Upload the binary to a release directory for tagged releases
- Adjust the docs to encourage people to use tagged releases instead of latest
- I also manually uploaded the tagged releases for past releases back to 3.43.0

Closes https://github.com/sourcegraph/sourcegraph/issues/42722

## Test plan

Will trigger a test CI run to validate the pipeline works.

### CODE REVIEW

File: doc/admin/deploy_executors_binary.md [hunk 1]
```diff
@@ -67,7 +67,10 @@ you run the latest version of Sourcegraph.
 Download and setup the `executor` binary:
 
 ```bash
-curl -sfLo executor https://storage.googleapis.com/sourcegraph-artifacts/executor/latest/linux-amd64/executor
+# Plug in the version of your Sourcegraph instance, like v4.1.0.
+# Before Sourcegraph 3.43.0, tagged releases are not available and you should default to using "latest" instead.
+# Using latest is NOT recommended, because it might be incompatible with your Sourcegraph version.
```
Comment (efritz) [inline doc/admin/deploy_executors_binary.md:72]: Maybe link them to docs to upgrade to a newer version?

Comment (eseliger) [reply doc/admin/deploy_executors_binary.md:72]: I don't have upgrade docs yet, open ticket for this sprint 😬

**Review by efritz — APPROVED** [review]
Thanks for the additional logging.

File: doc/admin/deploy_executors_binary.md (hunk 1 above)
Comment (efritz) [reply doc/admin/deploy_executors_binary.md:72]: I meant instance upgrade docs if they're behind 3.43.

Comment (eseliger) [reply doc/admin/deploy_executors_binary.md:72]: Ah!

File: doc/admin/deploy_executors_binary.md [hunk 2]
```diff
@@ -67,7 +67,10 @@ you run the latest version of Sourcegraph.
 Download and setup the `executor` binary:
 
 ```bash
-curl -sfLo executor https://storage.googleapis.com/sourcegraph-artifacts/executor/latest/linux-amd64/executor
+# Plug in the version of your Sourcegraph instance, like v4.1.0.
+# Before Sourcegraph 3.43.0, tagged releases are not available and you should default to using "latest" instead.
```
Comment (mrnugget) [inline doc/admin/deploy_executors_binary.md:71]:
```suggestion
# Before Sourcegraph 3.43.0, tagged releases of executors are not available and you should default to using "latest" instead.
```

**Review by mrnugget — APPROVED** [review]

**Review by Piszmog — APPROVED** [review]

**Pull request merged**